### PR TITLE
fix: ensure pubkey ante handler works as expected

### PR DESCRIPTION
**Pull request by PaddyMc**
## What is the purpose of the change

Test the public key ante handler and fix issue with setting incorrect public key

## Testing and Verifying

Added a test case to ensure issue is fixed

- `cd /x/authenticator/ante`
- `go test -v -cover ./...`

### CODE REVIEW

File: x/authenticator/ante/fees.go [hunk 1]
```diff
@@ -1 +0,0 @@
-package ante
```
Comment (nicolaslara) [inline x/authenticator/ante/fees.go:1]: can we remove this file?

File: x/authenticator/ante/pubkey.go [hunk 2]
```diff
@@ -1,9 +1,7 @@
 package ante
 
-// TODO: This is just a fork of the sdk's SetPubKeyDecorator. Here I'm allowing any pubkey so I can test the ante.
```
Comment (nicolaslara) [inline x/authenticator/ante/pubkey.go:3]: might still be worth it to have a comment explaining the diffs between this one and the default. Useful for future devs

**Review by nicolaslara — APPROVED** [review]

File: x/authenticator/ante/fees.go (hunk 1 above)
Comment (PaddyMc) [reply x/authenticator/ante/fees.go:1]: It's removed :muscle:

File: x/authenticator/ante/pubkey.go (hunk 2 above)
Comment (PaddyMc) [reply x/authenticator/ante/pubkey.go:3]: updated the comment explaining

**Pull request merged**